docs/DeveloperGuide.md

```diff
@@ -252,6 +252,80 @@ _{more aspects and alternatives to be added}_
+
+* **Alternative 1 (current choice):** Only title is required when creating a task. The other fields are optional.
+    * Pros: As users may not have a content or status in mind when creating new tasks, this alternative allows flexibility
+  in user input, which makes the app more user-friendly. Some tasks may also be self-explanatory and thus do not 
+  require content descriptions.
+    * Cons: More difficult to implement, more likely to cause bugs.
+
+* **Alternative 2:** Require all fields to be compulsory
+    * Pros: Easier to implement, fewer bugs may be generated.
+    * Cons: Less intuitive and less user-friendly, as users who might not have a content description in mind when 
+  creating tasks may be forced to key in random content to add tasks.
+
+#### Constraints:
+**Title must be unique:** </br>
+We felt that the title should be unique as it improves organisation and visual clarity for the user. By mandating unique 
+titles, we encourage users to be specific in the title(purpose) of the task (e.g they will set title as 
+"Complete slides for Mr X" rather than "Complete Slides"), which will benefit them greatly, as they will be 
+able to clearly distinguish the purpose of each task just by looking at the title. </br>
+Suppose that the title was not unique. Users might have many tasks with the same title, which would impair their ability
+to distinguish between the tasks unless they read each of the task content individually. It would also impair visual
+clarity when searching for tasks, as tasks with similar titles might clutter up the GUI.
+Hence, our approach in mandating unique titles are geared towards improving organisation and visual clarity for users in 
+both the short and long term.
+
+### Deleting a task
+Syntax: `deletetask INDEX` </br>
+Purpose: Allows users to delete the task at the specified index in the OfficeConnectModel.
+
+#### Implementation
+The implementation of this feature is supported by `ListTaskCommand`, `DeleteTaskCommand` and `DeleteTaskCommandParser`.
+Below are the steps required to delete a task in the OfficeConnectModel. </br>
+
+Step 1: User keys in `listtask`, which will display the index of all tasks. 
+The user can thus obtain the index of the task that they want to delete.
+
+Step 2: User keys in `deletetask INDEX` to delete the task at the specified index. 
+If the index is invalid, an error will be thrown. 
+
+Below is an activity diagram showcasing the 2 steps: </br> 
+![DeleteTaskActivityDiagram](images/DeleteTaskActivityDiagram.png)
+
+#### Design Considerations
+**Aspect: Implementation of Delete Task Command**
+
+* **Alternative 1 (current choice):** Require users to call `listtask` to get the index of the task they wish to delete.
+    * Pros: Increase convenience for users, as they do not have to remember the index of each task. Also easier to implement.
+    * Cons: Increases coupling within OfficeConnectModel, as `deletetask` requires `listtask` to function properly.
```

Comment:
I am not sure about this, because users technically don't have to call `listtask` to get the index, they can delete tasks as long as there are tasks present in the FilteredList. Maybe you can rephrase this?

Reply:
okay ill try to rephrase it more clearly